src/flux/scores.coffee

```diff
@@ -10,105 +11,111 @@ REJECTING = 'rejecting'
 REJECTED = 'rejected'
 
 
-
 allStudents = (scores) ->
   _.chain(scores)
     .pluck('students')
     .flatten(true)
     .value()
 
+getTaskInfoById = (taskId, data) ->
+  taskId = parseInt(taskId, 10)
+  for courseId, period of data
+    for period, periodIndex in data[courseId]
+      for student, studentIndex in period.students
+        for task in student.data
+          if task.id is taskId
+            return {task, courseId, period, periodIndex, studentIndex}
```

Comment:
If a class has 200 tasks and 500 students, is this nested iteration going to be too slow?  If so, do a one-time, possibly-laze build of a hash mapping `taskId` to `{task, courseId, period, periodIndex, studentIndex}`?